### PR TITLE
Feature/UI layout changes

### PR DESCRIPTION
**Pull request by simoncollins**
Got basic question and response layout working. Also snuck in support for markdown rendering including GFM and added nice styling of response HTML with tailwind's typography package:

image

image

image

image

Added progress spinner:

image

- Reactored app.tsx into separate components and introduced Zustand for state management
- Pull some logic out into custom hooks.

Known issues / remaining work:

- Response area doesn't scroll if text overflows
- Settings and chat buttons in top right do nothing except pop up alert box
- Need to remove focus from text area and shrink font size when query is running and other little UX improvements
- Need to reset to question mode when the user clicks into the question area again
- Need to add copy button to copy response to clipboard
- Need to add support for syntax highlighting of code.
- Need to make the response text styling better - should be inverted colors to match dark mode

### CODE REVIEW

File: src/App.tsx
Comment (Dax911) [inline src/App.tsx:1]: Heck yes thank you for breaking it out into components

**Review by Dax911 — COMMENTED** [review]
Seems fine to me, can u justify why u wanted to use zustand for this esp since the context and state is already so tightly bound to the rust backend. I mean its a great pattern and stuff just wanna know ur thought process on this.

**Review by sammcj — DISMISSED** [review]
LGTM

**Comment by simoncollins** [comment]
Hey @Dax911,

> can u justify why u wanted to use zustand for this esp since the context and state is already so tightly bound to the rust backend

Good question and I did mull over whether it was worth it. I initially implemented it without but found that I was prop drilling a bit to use state in different component subtrees once I split stuff from `App` out into different component, so I figured it was worth it for simplifying the logic without adding too much overhead.

Definitely a good question about the split between Rust side and Tauri side state though.

**Comment by sammcj** [comment]
linting will pass after this is merged - https://github.com/ai-qol-things/rusty-ollama/pull/35

**Review by sammcj — APPROVED** [review]

**Pull request merged**